### PR TITLE
CE-1844 Use flag_type_id as a key when fetching flags for a page

### PR DESCRIPTION
**Pull request by adamkarminski**
@lukaszkonieczny

### CODE REVIEW

File: extensions/wikia/Flags/controllers/FlagsController.class.php [hunk 1]
```diff
@@ -45,21 +45,88 @@ private function processRequest() {
 	 */
 
 	/**
-	 * Retrieves all data for flags assigned to the given page.
+	 * Retrieves all data for all flag types available on the given wikia
+	 * with an intent of rendering a modal with an edit form.
+	 * It returns
+	 * @see getFlagsForPageForEdit()
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: this is actually getFlagsForPageForEdit documentation, so this line is unnecessary?

File: extensions/wikia/Flags/controllers/FlagsController.class.php [hunk 2]
```diff
@@ -45,21 +45,88 @@ private function processRequest() {
...
+	 *
+	 * 		int flag_type_id
+	 * 		int wiki_id
+	 *		int page_id
+	 * 		int flag_group
+	 * 		string flag_name
+	 * 		string flag_view
+	 * 		int flag_targeting
+	 * 		string|null flag_params_names
+	 *
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: There's one more param in result (flag_view_url), right?

File: extensions/wikia/Flags/controllers/FlagsController.class.php [hunk 3]
```diff
@@ -45,21 +45,88 @@ private function processRequest() {
...
+		 * 5. Set the response data
+		 */
+		$this->setResponseData( $responseData );
+	}
+
+	/**
+	 * Retrieves all data for flags assigned to the given page
+	 * with an intent of rendering them. To get all types of flags:
+	 * @see getFlagsForPageForEdit()
+	 *
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: What about leaving just @see getFlagsForPageForEdit and removing below list and marking just the difference?

File: extensions/wikia/Flags/models/Flag.class.php [hunk 4]
```diff
@@ -29,8 +29,8 @@ public function getFlagsForPage( $wikiId, $pageId ) {
 
 		if ( !empty( $flagsWithParams ) ) {
 			$flagsParams = $this->getParamsForFlags( $flagsWithParams );
-			foreach ( $flagsParams as $flagId => $flagParams ) {
-				$flagsForPage[ $flagId ]['params'] = $flagParams;
+			foreach ( $flagsParams as $flagTypeId => $flagParams ) {
+				$flagsForPage[ $flagTypeId ]['params'] = $flagParams;
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/models/Flag.class.php]: Per coding conventions no spacing needed around $flagTypeId

File: extensions/wikia/Flags/models/Flag.class.php [hunk 5]
```diff
@@ -51,7 +51,7 @@ private function getFlagsTypesForPage( $wikiId, $pageId ) {
 			->WHERE( 'flags_to_pages.wiki_id' )->EQUAL_TO( $wikiId )
 			->AND_( 'flags_to_pages.page_id' )->EQUAL_TO( $pageId )
 			->runLoop( $db, function( &$flagsTypes, $row ) {
-				$flagsTypes[ $row->flag_id ] = get_object_vars( $row );
+				$flagsTypes[ $row->flag_type_id ] = get_object_vars( $row );
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/models/Flag.class.php]: No spacing needed around array key

File: extensions/wikia/Flags/models/Flag.class.php (hunk 4 above)
Comment (adamkarminski) [inline extensions/wikia/Flags/models/Flag.class.php]: Used for readability

File: extensions/wikia/Flags/models/Flag.class.php [hunk 6]
```diff
@@ -61,11 +61,11 @@ private function getParamsForFlags( $flagsWithParams ) {
 		$db = $this->getDatabaseForRead();
 
 		$flagsParams = ( new \WikiaSQL() )
-			->SELECT( 'flag_id', 'param_name', 'param_value' )
+			->SELECT( 'flag_type_id', 'param_name', 'param_value' )
 			->FROM( 'flags_params' )
 			->WHERE( 'flag_id' )->IN( $flagsWithParams )
 			->runLoop( $db, function( &$flagsParams, $row ) {
-				$flagsParams[ $row->flag_id ][ $row->param_name ] = $row->param_value;
+				$flagsParams[ $row->flag_type_id ][ $row->param_name ] = $row->param_value;
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/models/Flag.class.php]: No spacing needed around array key

File: extensions/wikia/Flags/models/FlagType.class.php [hunk 7]
```diff
@@ -23,7 +23,17 @@ public function getFlagGroupsMapping() {
...
 
 	public function getFlagTypesForWikia( $wikiId ) {
+		$db = $this->getDatabaseForRead();
 
+		$flagTypesForWikia = ( new \WikiaSQL() )
+			->SELECT_ALL()
+			->FROM( self::FLAGS_TYPES_TABLE )
+			->WHERE( 'wiki_id' )->EQUAL_TO( $wikiId )
+			->runLoop( $db, function( &$flagTypesForWikia, $row ) {
+				$flagTypesForWikia[ $row->flag_type_id ] = get_object_vars( $row );
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/models/FlagType.class.php]: No spacing needed around array key

File: extensions/wikia/Flags/controllers/FlagsController.class.php (hunk 3 above)
Comment (adamkarminski) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: I find it non-practical when reading the docs

File: extensions/wikia/Flags/controllers/FlagsController.class.php (hunk 2 above)
Comment (adamkarminski) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: True

File: extensions/wikia/Flags/controllers/FlagsController.class.php (hunk 1 above)
Comment (adamkarminski) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: Oops, you've found an unfinished part. I've corrected it and will push it with the next PR.

File: extensions/wikia/Flags/controllers/FlagsController.class.php [hunk 8]
```diff
@@ -45,21 +45,88 @@ private function processRequest() {
...
+	 * 	@if flag_params_names is not empty
+	 * 		params = [
+	 * 			param_name => param_value
+	 *		]
+	 */
+	public function getFlagsForPageForEdit() {
+		$this->getRequestParams();
+
+		if ( !isset( $this->params['pageId'] ) ) {
```
Comment (kamilkoterba) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: Can we throw something here to be more verbose?

File: extensions/wikia/Flags/models/Flag.class.php (hunk 4 above)
Comment (kamilkoterba) [inline extensions/wikia/Flags/models/Flag.class.php]: Perhaps you should raise that idea among engineering and update conventions here https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#array-syntax

File: extensions/wikia/Flags/controllers/FlagsController.class.php (hunk 8 above)
Comment (adamkarminski) [inline extensions/wikia/Flags/controllers/FlagsController.class.php]: To be done with general error handling.

**Comment by kamilkoterba** [comment]
Some minor comments, otherwise LGTM
Don't think any of these are blockers if you want to address them later

**Pull request merged**